### DISCUSSION TITLE
TTS Hebrew?!

### DISCUSSION THREAD

**Discussion by openfnord** [osmandapp/OsmAnd#13631] [Maps Related]
Dear People, thanks for osmand, it is really good. 
When I travel Israel, I some times need to use wayze,
as Israelis use it for good coverage and the nice hebrew
voices.
So I would like to request a hebrew text to speech voice
for osmand, as this is critical for osmand to read the
street names correctly. So without hebrew it justs 
omits the street names.
So it defaults to english and says in Israel:
"turn left on \*silence\* street"
as I can not  talk hebrew!

I would expecpect something like 
"pnei jamina be rehov ha rav kook"

or in englisch
"keep left at ha rav kook street"
or not so nice but maybe easier to implement
(rehov ha rav kook is to be the full name of the street in this example)

"keep left at rehov ha rav kook street"
(rehov=street) which would be double, a bit incorrect but who cares :-)

or:
hebrew voice:
"Be Kikar Ha Shabat kach et ha iziah shlishit"

englisch:
"At Circle (Kikar) Ha Shabat take the third exit"

Thanks in advance.

**Comment by vshcherb** [comment]
There is an open issue to Switch TTS engines during talk though it seems pretty complicated cause TTS-engine initialization will take time.

**Reply by openfnord** [reply]
**warming up the tts-engines**
If you want to switch TTS it would normally only use
2 different TTS (your inteface language, e.g. French, English) 
and the local language (e.g. Hindi, Georgian, Hebrew, Arabic)
So it should be able to have bothe TTS engines running, and 
feeding the engines depending on the character range in 
unicode. So each word - depending on the unicode characters 
contained - of the two languages (UI vs. MAP) goes to the appropriate
TTS. Audio out of both TTS is simply added like in an audio mixer.
This should work at the expense of RAM and processing power.

**Deluxe-Use-Case:**
So one has 3 languages, and 3 TTS running at the same time maximum:
You are at a border with different languages, so you have MAP_lang1, MAP_lang2
at the border and your UI language.

**Hebrew is missing completely and is needed**
Ok, but having hebrew in the first place would help a lot,
as hebrew does not exist but a lot of other languages.
A non jewish tourist would learn some hebrew words
like kikar, jamin, jashar and so on and a jewish person
would more feel at home :-)

Having the local language always helps, as if you ask
someone about a street in english, it helps to be able
to phonetically reproduce the local street name.
So if osmand tells it to you it helps much.

**Reply by Lee-Carre** [reply]
> There is an open issue to Switch TTS engines during talk

#4190
